hazelcast/src/test/java/com/hazelcast/config/XmlConfigSchemaLocationTest.java

```diff
@@ -0,0 +1,164 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import static com.hazelcast.test.TestCollectionUtils.setOf;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class XmlConfigSchemaLocationTest extends HazelcastTestSupport {
+    private static final Set<String> WHITELIST = setOf("hazelcast-sample-service.xsd");
+
+    private static final String XML_SCHEMA_NAMESPACE = "http://www.w3.org/2001/XMLSchema-instance";
+    private static final String XML_SCHEMA_LOCATION_ATTRIBUTE = "schemaLocation";
+
+    private CloseableHttpClient client;
+    private DocumentBuilderFactory documentBuilderFactory;
+    private Set<String> urlCache;
+
+    @Before
+    public void setUp() {
+        client = HttpClients.createDefault();
+        documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+        urlCache = new HashSet<String>();
+    }
+
+    @After
+    public void tearDown() {
+        IOUtil.closeResource(client);
+    }
+
+    @Test
+    public void testSchemaLocationsExist() throws Exception {
+        ResourcesScanner scanner = new ResourcesScanner();
+        Reflections reflections = new Reflections(scanner);
+
+        Set<String> resources = reflections.getResources(Pattern.compile(".*\\.xml"));
+        ClassLoader classLoader = getClass().getClassLoader();
+        for (String resource : resources) {
+            URL resourceUrl = classLoader.getResource(resource);
+            String protocol = resourceUrl.getProtocol();
+
+            // do not validate schemas from JARs (libraries). we are interested in local project files only.
+            if (protocol.startsWith("jar")) {
+                continue;
+            }
+
+            InputStream stream = null;
+            try {
+                stream = classLoader.getResourceAsStream(resource);
+                validateSchemaLocationUrl(stream, resource);
+            } finally {
+                IOUtil.closeResource(stream);
+            }
+        }
+    }
+
+    private void validateSchemaLocationUrl(InputStream documentStream, String originalLocation) throws Exception {
+        String schemaAttr = extractSchemaAttribute(documentStream);
+        if (schemaAttr == null) {
+            return;
+        }
+        for (String nameSpaceUrl : schemaAttr.split(" ")) {
+            nameSpaceUrl = nameSpaceUrl.trim();
+            if (shouldSkipValidation(nameSpaceUrl)) {
```

Comment:
this is ugly. instead of parsing locations from "schemaUri location" it's just skipping everything what does not end with `.xsd`. it works in practice quite well.